keymap.c

```diff
@@ -195,6 +195,38 @@ static const struct Extkey ExtKeys[] = {
 };
 #endif
 
+/**
+ * mutt_keymap_free - Free a Keymap
+ * @param km Keymap to free
+ */
+static void mutt_keymap_free(struct Keymap **km)
+{
+  if (!km || !*km)
+    return;
+
+  FREE(&(*km)->macro);
+  FREE(&(*km)->desc);
+  FREE(&(*km)->keys);
+  FREE(km);
+}
+
+/**
+ * mutt_keymaplist_free - Free a List of Keymaps
+ * @param km_list List of Keymaps to free
+ */
+static void mutt_keymaplist_free(struct KeymapList *km_list)
+{
+  struct Keymap *np = NULL, *tmp = NULL;
+
+  STAILQ_FOREACH_SAFE(np, km_list, entries, tmp)
+  {
+    STAILQ_REMOVE(km_list, np, Keymap, entries);
+    mutt_keymap_free(&np);
+  }
+
+  STAILQ_INIT(km_list);
```

Comment:
So I think you're mixing two approaches here:

1. iterate, remove and free
```c
struct Keymap *np = NULL, *tmp = NULL;
STAILQ_FOREACH_SAFE(np, km_list, entries, tmp)
{
  STAILQ_REMOVE(km_list, np, Keymap, entries)
  mutt_keymap_free(&np);
}
```

and

2. iterate, free, adjust pointers at the end
```c
struct Keymap *np = STAILQ_FIRST(km_list), *next = NULL;
while (np)
{
  next = STAILQ_NEXT(np, entries);
  mutt_keymap_free(&np);
}
STAILQ_INIT(km_list);
```

By looking for `STAILQ_INIT`  in the code, it seems you're not the first one :)

Reply:
Fixed. Went with option 1.